limitedlistener.go

```diff
@@ -74,10 +77,16 @@ func (sl *limitedListener) Restart() {
 type LimitedConn struct {
 	net.Conn
 	counter *uint64
+	closed  uint32
 }
 
 func (c *LimitedConn) Close() (err error) {
+	if atomic.SwapUint32(&c.closed, 1) == 1 {
```

Comment:
Avoid decreasing `numConns` multiple times when closing, or it will become negative (huge `uint64` value actually) and make chaos.